draft-ietf-capport-architecture.xml

```diff
@@ -349,52 +348,16 @@
-        of its access to the external network, so providing the Enforcement Device the ability
-        to preemptively signal may be desirable.
-        </t>
-        <t>
-        A specific Captive Portal Signaling Protocol is out of scope for this document.
-        However, in order to ensure that future protocols fit into the architecture,
-        requirements for a Captive Portal Signaling Protocol follow:
+        When User Equipment first connects to a network, or when there are changes in status,
+        the Enforcement Device could generate a signal toward the User Equipment.  This signal
+        indicates that the User Equipment might need to contact the API Server to receive
+        updated information.  For instance, this signal might be generated when the end of a
+        session is imminent, or when network access was denied.
         </t>
-        <ol spacing="normal" type="1">
-          <li>The notification SHOULD NOT be easy to spoof.
-               If an attacker can send spoofed notifications to the User Equipment,
-               they can cause the User Equipment to unnecessarily access the API. Rather than
-               relying solely on rate limits to prevent problems, a good protocol will strive
-               to limit the feasibility of such attacks.
-            </li>
-          <li>It SHOULD be possible to send the notification before the captive portal closes.
-               This will help ensure seamless connectivity for the user, as the User Equipment
-               will not need to wait for a network failure to refresh its login. On receipt of
-               preemptive notification, the User Equipment can prompt the user to refresh.
-             </li>
-          <li>The signal SHOULD NOT include any information other than an indication that traffic
-                is restricted, which can prompt the User Equipment to contact the API.</li>
-        </ol>
-        <t>
 
-       The Captive Portal Signaling Protocol does not provide any means of indicating that the
-       network prevents access to some destinations. The intent is to rely on the Captive Portal API
-       and the web portal to which it points to communicate local network policies.
-        </t>
-        <t>The Captive Portal Enforcement function MAY send Captive Portal Signals when disallowed
-	  User Equipment attempts to send to the network.  These signals MUST be rate-limited to a
-          configurable rate.
-        </t>
         <t>
-         The signals MUST NOT be sent to the destinations/peers that the User Equipment is
-         restricted from accessing.
-         The indications are only to be sent to the User Equipment.
+        An Enforcement Device MUST rate limit any signal generated in response to these conditions.  See <xref target="section_signal_risks"/> for a discussion of
+        risks related to a Captive Portal Signal.
```

Comment:
In Martin's example on the mailing list, he mentioned rate limiting API requests. We already mention this in the API section where it discusses the captive portal signal.

Reply:
Wasn't it supposed to be the user device that rate-limits responses to the captive portal signal?

Reply:
Yes. The UE should rate limit any requests to the API in response to the signal. See https://github.com/capport-wg/architecture/blob/master/draft-ietf-capport-architecture.xml#L311 (which I haven't changed here)